Makefile

```
@@ -14,7 +14,10 @@ else
 
 # This is the "normal" part of the Makefile
 
-TAR := $(shell hash bsdtar > /dev/null 2>&1 && echo "bsdtar --no-xattrs" || echo "tar" )
+TAR := $(shell hash bsdtar > /dev/null 2>&1 && echo "bsdtar" || echo "tar" )
+ifneq ($(shell $(TAR) --version | grep "bsdtar 3"),)
+	TAR += --format=gnutar
+endif
```

I vote we do not have any more bsdtar stuff here. By all means allow bsdtar users to set an environment variable to use it like this:

```suggestion
TAR ?= tar
```

So that if you run: `TAR="bsdtar --no-xattrs --format=gnutar" make ...` you use it.

I'd even state that this:

```
$(eval EXCL := --exclude=$(shell [ ! "$(TAR)" = "tar" ] && echo "^" )./)
```

needs to go too but I'm not averse to hacks like detecting if the provided tar is bsdtar e.g.

```
$(eval EXCL := --exclude=$(shell [ "$(TAR)" = bsdtar* ] && echo "^" )./)
```

Why are we even still going through this - what are the real benefits to using bsdtar? if you're going to use the --format=gnutar does it even have any improvement anymore?

---

It started because macOS doesn't ship GNU Tar, so the original script led to wrong files being included (which your new fix doesn't handle). The last PR was very close to a good solution already. I'm not against removing it from `.drone.yml` anyway.

`bsdtar`/libarchive still comes with all the benefits with file handling and compression and the like when using a different output stream writer, and as stated the (good) compatibility of these archives are documented clearly.

---

Cf. relevant change suggested to #15273 for fixing Mac issue singularly.